h2o-core/src/main/java/water/util/FrameUtils.java

```diff
@@ -111,6 +111,37 @@ public static Frame categoricalEncoder(Frame dataset, String[] skipCols, Model.P
     }
   }
 
+  /**
+   * Helper method to apply TargetEncoderModel
+   * @param teModel trained TargetEncoderModel
+   * @param inputFrame frame to apply target encoding to
+   * @param scopeTrack `true` if Scope is tracking keys
+   * @return a Frame with encodings
+   */
+  public static Frame internal_applyTargetEncoder(Model teModel, Frame inputFrame, boolean scopeTrack) {
```

Comment:
This belongs somewhere close to Target Encoder package, not to generate frame utils. This way, anything could be utils.

Reply:
Agree that this should go somewhere else.
Can't go to TE module though as it would create a dependency from core to targetencoder.
Let me think about some abstraction, even if temporary, that will allow to group all this logic to the TE module.